octree/include/pcl/octree/octree_container.h

```diff
@@ -89,21 +89,22 @@ class OctreeContainerBase {
   /** \brief Empty addPointIndex implementation. This leaf node does not store any point
    * indices.
    */
-  void
-  addPointIndex(const index_t&)
-  {}
+  virtual void addPointIndex(index_t) {}
 
   /** \brief Empty getPointIndex implementation as this leaf node does not store any
    * point indices.
    */
-  void
-  getPointIndex(index_t&) const
```

Comment:
Simply removing this might be problematic since people may use it. How about just adding the new virtual function?